tests/entrypoint_hook.py

```diff
@@ -0,0 +1,120 @@
+import sys
+import os
+import json
+import difflib
+import entrypoint
+
+class CommandNotFound(Exception):
+    """Raised when entrypoint command is not found or test fail."""
+
+class Command:
+    """
+    Represent a single execve command, with
+    its arguments and environment.
+
+    Can represent an entrypoint hooked command, the expected
+    result or the input of a test.
+    """
+    def __init__(self, argv, environ):
+        #sort cli arguments to facilitate comparaison
+        sorted_args = argv[1:]
+        sorted_args.sort()
+        self.argv = [argv[0]] + sorted_args
+        self.environ = environ
+
+    def __eq__(self, other):
+        """Compare 2 Command, result of a test and expected command."""
+        return self.argv == other.argv and self.environ == other.environ
+
+    def stringify(self):
+        """Render single command into string for error outputs."""
+        argv_str = json.dumps(self.argv, indent=4)
+        command_str = f"argv: {argv_str}\n"
+        environ_str = json.dumps(self.environ, indent=4)
+        command_str += f"environ: {environ_str}"
+        return command_str
+
+    def __repr__(self):
+        """Override pytest assert message error"""
+        return "<420UselessMessage420>"
```

Comment:
It was for the error message on test failure, to override pytest default message. It was dirty because I didn't know how to deal with it properly. Fixed, I changed the way to display error message, using [pytest_assertrepr_compare](https://docs.pytest.org/en/latest/how-to/assert.html#defining-your-own-explanation-for-failed-assertions).